packages/common/src/location/util.ts

```diff
@@ -38,10 +38,13 @@ export function joinWithSlash(start: string, end: string) {
  * @returns The URL string, modified if needed.
  */
 export function stripTrailingSlash(url: string): string {
-  const match = url.match(/#|\?|$/);
-  const pathEndIdx = (match && match.index) || url.length;
-  const droppedSlashIdx = pathEndIdx - (url[pathEndIdx - 1] === '/' ? 1 : 0);
-  return url.slice(0, droppedSlashIdx) + url.slice(pathEndIdx);
+  // Find the index of the first occurrence of `#`, `?`, or the end of the string.
+  // This marks the start of the query string, fragment, or the end of the URL path.
+  const pathEndIdx = url.search(/#|\?|$/);
```

Comment:
Would this be better?
```suggestion
  const pathEndIdx = url.search(/[#?$]/);
```

Reply:
That would be another regex, I just took the existing one to avoid any breaking changes.